README.md

```diff
@@ -1,207 +1,607 @@
+| type              | string                                    | The media type of the artifact (see [Model Artifact Media-Type](#model-artifact-media-type).     |
+| roles             | \[string]                                 | **REQUIRED** Specify `mlm:model`. Can include `["mlm:weights", "mlm:checkpoint"]` as applicable. |
+| mlm:artifact_type | [Artifact Type Enum](#artifact-type-enum) | Specifies the kind of model artifact. Typically related to a particular ML framework.            |
+
+Recommended Asset `roles` include `mlm:weights` or `mlm:checkpoint` for model weights that need to be loaded by a
+model definition and `mlm:compiled` for models that can be loaded directly without an intermediate model definition.
+In each case, the `mlm:model` should be applied as well to indicate that this asset represents the model.
+
+It is also recommended to make use of the
+[file](https://github.com/stac-extensions/file?tab=readme-ov-file#asset--link-object-fields)
+extension for this Asset, as it can provide useful information to validate the contents of the model definition,
+by comparison with fields `file:checksum` and `file:size` for example.
+
+#### Model Artifact Media-Type
+
+Not all ML framework, libraries or model artifacts provide explicit media-type. When those are not provided, custom
+media-types can be considered. For example `application/x-pytorch` or `application/octet-stream; application=pytorch`
+could be appropriate to represent a PyTorch `.pt` file, since the underlying format is a serialized pickle structure.
+
+#### Artifact Type Enum
+
+This value can be used to provide additional details about the specific model artifact being described.
+For example, PyTorch offers various strategies for providing model definitions, such as Pickle (`.pt`), TorchScript,
+or the compiled approach. Since they all refer to the same ML framework,
+the [Model Artifact Media-Type](#model-artifact-media-type) would be insufficient in this case to detect with strategy
+should be used. 
+
+Following are some proposed *Artifact Type* values for corresponding approaches, but other names are
+permitted as well. Note that the names are selected using the framework-specific definitions to help
+the users understand the source explicitly, although this is not strictly required either.
+
+| Artifact Type      | Description                                                                                                              |
+|--------------------|--------------------------------------------------------------------------------------------------------------------------|
+| `torch.compile`    | A model artifact obtained by [`torch.compile`](https://pytorch.org/tutorials/intermediate/torch_compile_tutorial.html).  |
+| `torch.jit.script` | A model artifact obtained by [`TorchScript`](https://pytorch.org/docs/stable/jit.html).                                  |
```

Comment:
```suggestion
| `torchscript` | A model artifact obtained by [`TorchScript Scripting`](https://pytorch.org/docs/stable/jit.html) and/or [`TorchScript Tracing`](https://pytorch.org/docs/stable/generated/torch.jit.trace.html).                                  |
```

There are two types of graph capture in torchscript, trace and script. I think we can either enumerate both or leave only one option. Either traced or scripted models can be loaded the same way so I favor only one field for both of them.

Somewhat confusingly, both can also be used together though this is not common. https://ppwwyyxx.com/blog/2022/TorchScript-Tracing-vs-Scripting/

Since Torchscript is bring phased out in favor of AOTInductor, I think we shouldn't make this too complex and only provide them as one field.

Reply:
Similar to other comment about the export. I'm not 100% convinced about this. I got somewhat mixed feelings about it. 
I find that `torchscript` might just make it more ambiguous by not being explicit (about the actual function) whether it refers to `torch.jit.trace` or `torch.jit.script`, but then, it might also just be a technicality about pytorch that we should omit for MLM since this table is already extremely PyTorch-opinionated.  

Also, I would like to remind that, even if things are being phased out, it does not mean we can ignore them. Some code/artifacts exists already, and users might what to represent them with the corresponding old technology.